rules/aip0214/resource_expiry.go

```diff
@@ -23,7 +23,14 @@ import (
 var resourceExpiry = &lint.FieldRule{
 	Name: lint.NewRuleName(214, "resource-expiry"),
 	OnlyIf: func(f *desc.FieldDescriptor) bool {
-		return f.GetName() == "expire_time"
+		// only return true if the parent is a resource
+		if utils.GetResourceFromDescriptor(f.GetParent()) == nil {
+			return false
+		}
+		if f.GetName() != "expire_time" {
+			return false
+		}
+		return true
```

Comment:
```suggestion
		isResource := utils.GetResourceFromDescriptor(f.GetParent()) != nil
		isExpireTime := f.GetName() == "expire_time"
		return isResource && isExpireTime
```

Kind of a nit, more my preference: It is easier to understand what the conditions are when testing for the positive case i.e. "Lint this only if parent is a resource and the field name is expire_time"

Reply:
I prefer the other case slightly (guard conditions are a little easier to follow for me), but not strong enough that I won't take your suggestion. Thanks!